### DISCUSSION TITLE
Google Analytics

### DISCUSSION THREAD

**Discussion by imakihi** [TerriaJS/terriajs#5615] [Q&A] [answered]
Hi Terria community!

I am trying to set up GA to monitor our Terria instance usage. 

Can anyone tell me how to set "googleAnalyticsKey"? Should I use a **client ID** described in this [site](https://www.rsjoomla.com/support/documentation/rsseo/troubleshooting/how-to-get-google-analytics-api-keys.html)?

Terria doc says
>  "A Google API key for Google Analytics. If specified, TerriaJS will send various events about how it's used to Google Analytics." 
What kind of various events I can monitor?

I also want to figure out what kind of "googleAnalyticsOptions" I can set? 

Sorry about many questions. If someone can share some config example, that would be super helpful!!!

Thank you in advance.

Hiroo

**Comment by hardreddata** [comment]
Hi @imakihi 

You pop `googleAnalyticsKey: "ABC123",` into your `wwwroot/config.json` file.

I am not sure about the options.

**Reply by imakihi** [reply]
Hi @RussellGrew. Thank you as usual!
I was asked what kind of stats we can get with this. So, I am going to try it.
Again, Thank you!

Best regards.

**Comment by AnaBelgun** [comment, marked as answer]
Hi @imakihi usually you can get by default nr of users, sessions, top events, referrals, etc.

**Reply by imakihi** [reply]
@AnaBelgun ! Thank you for the info. I somehow set  GA and I am getting some inputs. Do you know how I can use "googleAnalyticsOptions"??

Thank you!

Hiroo

**Reply by imakihi** [reply]
@AnaBelgun @RussellGrew Thank you! I could set GA on our Terria instance. Now I can analyze all user activities. I especially like that I can see which datasets are popular among our users. There are lots of events I can monitor improve our services.
This is awesome.
Thanks again.
